src/main/java/software/amazon/awssdk/crt/io/TlsContextOptions.java

```diff
@@ -207,8 +215,10 @@ public void overrideDefaultTrustStoreFromPath(String caPath, String caFile) {
      * @param caRoot Buffer containing the root certificate chain. Must be in PEM format.
      */
     public void overrideDefaultTrustStore(String caRoot) throws IllegalArgumentException {
-        caRoot = PemUtils.removeInvalidPemChars(caRoot);
-        caRoot = PemUtils.cleanUpPem(caRoot);
+        String cleanCaRoot = PemUtils.cleanUpPem(caRoot);
+        if (!cleanCaRoot.equals(caRoot)) {
+            throw new IllegalArgumentException("Trust Store contains excess or illegal characters");
```

Comment:
CA files on CentOS host contain comments, this change will break them. 

See: 
 - https://github.com/awslabs/s2n/issues/1136
 - https://gist.github.com/alexw91/502f439535bf9886851d0ba9ce27043a

Reply:
Gah, good to know. I'm tempted to just clean stuff up and call it good then. When we have a mechanism for logging from the Java side of the CRT, we can add a warning if we want.